src/frontend/lib/file-system.ts

```diff
@@ -28,17 +28,22 @@ export async function selectFile(opts: {
     throw new Error();
   }
 
-  const hasValidExtension = opts.extensionFilters
-    ? opts.extensionFilters.some(extension =>
-        asset.uri.endsWith(`.${extension}`),
-      )
-    : true;
-
-  if (!hasValidExtension) {
-    FileSystem.deleteAsync(asset.uri).catch(err => {
-      console.log(err);
+  if (opts.extensionFilters) {
+    const hasValidExtension = opts.extensionFilters.some(extension => {
+      return asset.name.endsWith(`.${extension}`);
```

Comment:
just highlighting the subtle change here. using the `name` is more reliable and agnostic to the `copyToCache` option because the uri only contains the original extension when copying to the cache.